libs/s25main/lua/LuaPlayer.cpp

```diff
@@ -259,15 +261,22 @@ bool LuaPlayer::AIConstructionOrder(unsigned x, unsigned y, lua::SafeEnum<Buildi
     return true;
 }
 
+void LuaPlayer::PlaceHQ(MapCoord x, MapCoord y)
+{
+    // Ignore if there is an HQ set in the map file.
+    if(player.GetHQPos().isValid())
+        return;
+
+    GameWorld& world = player.GetGameWorld();
+    const MapPoint mp{x, y};
+    constexpr auto checkExists = false;
+    world.DestroyNO(mp, checkExists);
+    BuildingFactory::CreateBuilding(world, BuildingType::Headquarters, mp, player.GetPlayerId(), player.nation);
```

Comment:
Nit: Better use `pos(ition)` as in other places

```suggestion
    const MapPoint pos{x, y};
    constexpr auto checkExists = false;
    world.DestroyNO(pos, checkExists);
    BuildingFactory::CreateBuilding(world, BuildingType::Headquarters, pos, player.GetPlayerId(), player.nation);
```